src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceParameterService.cs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        protected override async Task<Solution> RewriteCallSitesAsync(ExpressionSyntax expression, ImmutableDictionary<Document, List<InvocationExpressionSyntax>> callSites,
+         IMethodSymbol methodSymbol, CancellationToken cancellationToken)
+        {
+            var mappingDictionary = TieExpressionToParameters(expression, methodSymbol);
+            expression = expression.TrackNodes(mappingDictionary.Values);
+            var identifiers = expression.DescendantNodes().Where(node => node is IdentifierNameSyntax);
+
+            if (!callSites.Keys.Any())
+            {
+                return null;
+            }
+
+            var firstCallSite = callSites.Keys.First();
+
+            var currentSolution = firstCallSite.Project.Solution;
+            foreach (var keyValuePair in callSites)
+            {
+                var document = currentSolution.GetDocument(keyValuePair.Key.Id);
+                var invocationExpressionList = keyValuePair.Value;
+                var generator = SyntaxGenerator.GetGenerator(document);
+                var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+                var invocationSemanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var editor = new SyntaxEditor(await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false), generator);
+
+                foreach (var invocationExpression in invocationExpressionList)
+                {
+                    var newArgumentExpression = expression;
+                    var invocationArguments = invocationExpression.ArgumentList.Arguments;
+                    foreach (var argument in invocationArguments)
+                    {
+                        var associatedParameter = semanticFacts.FindParameterForArgument(invocationSemanticModel, argument, cancellationToken);
+                        var parenthesizedArgumentExpression = generator.AddParentheses(argument.Expression, false);
+                        if (!mappingDictionary.TryGetValue(associatedParameter, out var value))
+                        {
+                            continue;
+                        }
+
+                        newArgumentExpression = newArgumentExpression.ReplaceNode(newArgumentExpression.GetCurrentNode(value), parenthesizedArgumentExpression);
+                    }
+                    var allArguments = invocationExpression.ArgumentList.Arguments.Add(SyntaxFactory.Argument(newArgumentExpression.WithoutAnnotations(ExpressionAnnotationKind).WithAdditionalAnnotations(Simplifier.Annotation)));
+                    editor.ReplaceNode(invocationExpression, editor.Generator.InvocationExpression(invocationExpression.Expression, allArguments));
+                }
+
+                var newRoot = editor.GetChangedRoot();
+                document = document.WithSyntaxRoot(newRoot);
+                currentSolution = document.Project.Solution;
```

Comment:
dont' modify solution in place.  it will make everything more expensive

Reply:
Will message you about this for alternatives and further understanding.